README.md

```diff
@@ -157,6 +157,26 @@ We currently have CI in-place to call the above script whenever a new provider v
 
 If you are looking to generate resource docs for _all_ providers, then you might find `./scripts/gen_all_resource_docs.sh` useful. The only use-case for this script is if you need to regenerate docs for all supported providers out-of-band from CI on your local machine and then a PR for it.
 
+## Enlisting a new provider for docs
+
+### Generating docs for a new provider
+Since Pulumi uses CI to generate docs, there isn't a need to keep scripts up-to-date when docs for a new provider
+ need to be generated. Every provider repo uses similar `Makefile`  that contains calls to the
+  relevant scripts to generate docs for that provider when a new version is released.
+
+However, there may be times when you need to generate docs for all providers out-of-band and submit a PR for it. To
+ do that, ensure that the following files have all of the providers listed:
+* For NodeJS API docs: `scripts/run_typedoc.sh`
+* For Python API docs: `scripts/generate_python_docs.sh` and `tools/pydocgen/pulumi-docs.json`
+* For .NET API docs: `docfx/docfx.json`
+* For Resource docs: `scripts/generate_all_resource_docs.sh`
```

Comment:
Lower priority, but we really should consolidate these into a single place. Should we open an issue?

Reply:
Yes, definitely! I was thinking the exact thing while I was typing this out. But yeah this is the current situation. And yes, I'll open an issue for it.

Reply:
Created https://github.com/pulumi/docs/issues/3168 for this.